OneSignalSDK/onesignal/src/main/java/com/onesignal/OSSessionManager.java

```diff
@@ -0,0 +1,295 @@
+        JSONArray notificationsReceived = OutcomesUtils.getLastNotificationsReceivedData();
+        JSONArray notificationsIds = new JSONArray();
+
+        long attributionWindow = OutcomesUtils.getIndirectAttributionWindow() * 60 * 1_000L;
+        long currentTime = System.currentTimeMillis();
+        for (int i = 0; i < notificationsReceived.length(); i++) {
+            try {
+                JSONObject jsonObject = notificationsReceived.getJSONObject(i);
+                long time = jsonObject.getLong(OutcomesUtils.TIME);
+                long difference = currentTime - time;
+
+                if (difference <= attributionWindow) {
+                    String notificationId = jsonObject.getString(OutcomesUtils.NOTIFICATION_ID);
+                    notificationsIds.put(notificationId);
+                }
+            } catch (JSONException e) {
+                OneSignal.Log(OneSignal.LOG_LEVEL.ERROR, "From getting notification from array:JSON Failed.", e);
+            }
+        }
+
+        return notificationsIds;
+    }
+
+    // Get the current session based on state + if outcomes features are enabled.
+    @NonNull SessionResult getSessionResult() {
+        if (session.isDirect()) {
+            if (OutcomesUtils.isDirectSessionEnabled()) {
+                JSONArray directNotificationIds = new JSONArray().put(directNotificationId);
+                return SessionResult.Builder.newInstance()
+                        .setNotificationIds(directNotificationIds)
+                        .setSession(Session.DIRECT)
+                        .build();
+            }
+        } else if (session.isIndirect()) {
+            if (OutcomesUtils.isIndirectSessionEnabled()) {
+                return SessionResult.Builder.newInstance()
+                        .setNotificationIds(indirectNotificationIds)
+                        .setSession(Session.INDIRECT)
+                        .build();
+            }
+        } else if (OutcomesUtils.isUnattributedSessionEnabled()) {
+            return SessionResult.Builder.newInstance()
+                    .setSession(Session.UNATTRIBUTED)
+                    .build();
+        }
+
+        return SessionResult.Builder.newInstance()
+                .setSession(Session.DISABLED)
+                .build();
+    }
+
+    /**
+     * Attempt to override the current session before the 30 second session minimum
+     * This should only be done in a upward direction:
+     *   * UNATTRIBUTED can become INDIRECT or DIRECT
+     *   * INDIRECT can become DIRECT
+     *   * DIRECT can become DIRECT
+     */
+    void attemptSessionUpgrade() {
+        // We will try to override any session with DIRECT
+        if (OneSignal.getAppEntryState().isNotificationClick()) {
+            setSession(Session.DIRECT, directNotificationId, null);
+            return;
+        }
+
+        // We will try to override the UNATTRIBUTED session with INDIRECT
+        if (session.isUnattributed()) {
```

Comment:
sorry just want to check this case, if i have my app with an INDIRECT session with indirect notifications [id: "abcd"], close the app, and receive an Notification with notification id ["efgh"], if a open the app from the app icon no new session was triggered, the session will still de indirect for the previous session and continue to have the notificationsId as [id: "abcd"], not [id: "abcd", "efgh"] thats the desire flow?

Reply:
Right we should just have one notification id, the old one of id `abcd` from your example.